Doppler.AccountPlans/Encryption/EncryptionService.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Extensions.Options;
+
+namespace Doppler.AccountPlans.Encryption
+{
+    public class EncryptionService : IEncryptionService, IDisposable
+    {
+        private readonly IOptions<EncryptionSettings> _options;
+        private bool _initialized;
+        private RijndaelManaged _symmetricKey;
+        private Lazy<ICryptoTransform> _encryptor;
+        private Lazy<ICryptoTransform> _decryptor;
+
+        public EncryptionService(IOptions<EncryptionSettings> options)
+        {
+            _options = options;
+        }
+
+        public string DecryptAES256(string input)
+        {
+            if (string.IsNullOrEmpty(input))
+            {
+                return string.Empty;
+            }
+
+            EnsureInitialization();
+
+            var cipherTextBytes = Convert.FromBase64String(input);
+
+            using var memoryStream = new MemoryStream(cipherTextBytes);
+            using var cryptoStream = new CryptoStream(memoryStream, _decryptor.Value, CryptoStreamMode.Read);
+            var plainTextBytes = new byte[cipherTextBytes.Length];
+            var decryptedByteCount = cryptoStream.Read(plainTextBytes, 0, plainTextBytes.Length);
+            return Encoding.UTF8.GetString(plainTextBytes, 0, decryptedByteCount);
+        }
+
+        public string EncryptAES256(string input)
+        {
+            if (string.IsNullOrEmpty(input))
+            {
+                return string.Empty;
+            }
+
+            EnsureInitialization();
+
+            var plainTextBytes = Encoding.UTF8.GetBytes(input);
+
+            using var memoryStream = new MemoryStream();
+            using var cryptoStream = new CryptoStream(memoryStream, _encryptor.Value, CryptoStreamMode.Write);
+            cryptoStream.Write(plainTextBytes, 0, plainTextBytes.Length);
+            cryptoStream.FlushFinalBlock();
+            return Convert.ToBase64String(memoryStream.ToArray());
+        }
+
+        private void EnsureInitialization()
```

Comment:
Why are you using a Lazy initialization?

And it is two times lazy because _encryptor and _decryptor are also lazy.

Maybe there is a reason, but we need to understand it.